webui/portal/src/main/java/org/exoplatform/portal/application/localization/DefaultLocalePolicyService.java

```diff
@@ -47,6 +64,48 @@
  * @author <a href="mailto:mstrukel@redhat.com">Marko Strukelj</a>
  */
 public class DefaultLocalePolicyService implements LocalePolicy, Startable {
```

Comment:
The goal of this service (LocalePolicy) is to determine the locale, not to collect the locale context (LocaleContextInfo). You should instead create a new service (something like LocaleService) that will construct the locale context and then use the LocalePolicy service to calculate the locale to use.